src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,172 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+/**
+ * Represents a parser for user input given to Duke chatbot.
+ *
+ * @author Conrad
+ */
+public class Parser {
+
+    /**
+     * Parses the task number from the user input command into its <code>TaskList</code> index.
+     *
+     * @param userInput String input provided by user.
+     * @return An integer representing the index of the task in the <code>TaskList</code>.
+     * @throws DukeException If input is not of the type [command] [taskNumber]
+     */
+    public static int parseTaskNumber(String userInput) throws DukeException {
+        String[] parsedInput = userInput.split(" ");
+        String taskNumberString = parsedInput[1];
+        try {
+            return Integer.parseInt(taskNumberString) - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("   " + taskNumberString + " is not a valid task number.\n");
+        }
+
+    }
+
+    /**
+     * Parses the user input into a <code>Todo</code> task.
+     *
+     * @param userInput String input provided by user.
+     * @return An <code>Todo</code> task with the specified description.
+     */
+    public static Todo parseTodoTask(String userInput) {
+        String[] parsedUserResponse = userInput.split(" ");
+        String newTaskDescription = String.join(" ",
+                Arrays.copyOfRange(parsedUserResponse, 1, parsedUserResponse.length));
+        return new Todo(newTaskDescription);
```

Comment:
Maybe we could do something like this, assuming we are trying to obtain the todo description from the command given?
```java
String[] newTaskDescription = userInput.replace("/todo", "").strip();
```